### PR TITLE
update v1.3 dg

### PR DESCRIPTION
**Pull request by KxxMxxx**
updated logic component and proposed feature implementation

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -243,6 +243,53 @@ _{Explain here how the data encryption feature will be implemented}_
...
+
+* `add activity n/lunch a/KFC c/5.00`
+
+Step 2. User adds another activity without a cost:
+
+* `add activity n/go for a walk a/park`
+
+Step 3. User schedules the first activity into a day:
+
+* `schedule 1 st/1300 et/1400 d/1`
```
Comment (1nefootstep) [inline docs/DeveloperGuide.adoc]: the current implementation is `schedule activity 1 st/1300 et/1400 d/1`

File: docs/diagrams/OptimiseSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,88 @@
+@startuml
```
Comment (1nefootstep) [inline docs/diagrams/OptimiseSequenceDiagram.puml]: We don't have this command yet; is this is the planned implementation of the command?

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -243,6 +243,53 @@ _{Explain here how the data encryption feature will be implemented}_
...
+
+Step 4. User schedules the second activity into the same day:
+
+* `schedule 2 st/1300 et/1400 d/1`
```
Comment (1nefootstep) [inline docs/DeveloperGuide.adoc]: Similarly, `schedule activity 2 st/1300 et/1400 d/1`
Also, the first activity and second activity have the same start time and end time. Was that intended?

File: docs/diagrams/OptimiseSequenceDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+activate OptimiseBudgetCommand
+
+OptimiseBudgetCommand -> ModelManager : getFilteredItinerary()
+activate ModelManager
+
+ModelManager --> OptimiseBudgetCommand
+deactivate ModelManager
+
+OptimiseBudgetCommand -> Day : get()
+activate Day
```
Comment (1nefootstep) [inline docs/diagrams/OptimiseSequenceDiagram.puml]: `get()` calling Day here feels odd. I would think that `get()` is to get the particular day of interest from the filteredItinerary. the get should be pointing to the class List<Day>.

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -243,6 +243,53 @@ _{Explain here how the data encryption feature will be implemented}_
...
+
+Step 5. User executes `optimise budget` command. The activities in that day would be shown in a way such that the activities with the lowest costs are in focus.
+
+Shown below is a summary of the execution of the command.
+
+image::OptimiseSequenceDiagram.png[]
+
```
Comment (1nefootstep) [inline docs/DeveloperGuide.adoc]: Why is the focus on optimise budget command when the discussion is about Schedule version comparison. 

Shouldn't this be a command to view different versions of the same day or to view a day with activities with conflicting start/end times?

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (KxxMxxx) [reply docs/DeveloperGuide.adoc]: alright thanks

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (KxxMxxx) [reply docs/DeveloperGuide.adoc]: alright thanks

File: docs/diagrams/OptimiseSequenceDiagram.puml (hunk 4 above)
Comment (KxxMxxx) [reply docs/diagrams/OptimiseSequenceDiagram.puml]: good point, I should change it to the list class instead

File: docs/DeveloperGuide.adoc (hunk 5 above)
Comment (KxxMxxx) [reply docs/DeveloperGuide.adoc]: This feature is more of a combination of both, the comparison being the front end part and the optimisation being the backend. Could you perhaps suggest a better naming for this feature? Thanks!

Comment (1nefootstep) [reply docs/DeveloperGuide.adoc]: Oh. How about 

> Recommended Optimisation

 Feature

Comment (KxxMxxx) [reply docs/DeveloperGuide.adoc]: Thanks

**Pull request merged**